core/src/main/kotlin/org/evomaster/core/problem/rest/RestIndividual.kt

```diff
@@ -153,4 +179,101 @@ class RestIndividual (
         }
         return true
     }
+
+    fun getResourceCalls() : List<RestResourceCalls> = resourceCalls.toList()
+
+    /****************************** manipulate resource call in an individual *******************************************/
+    fun removeResourceCall(position : Int) {
+        if(position >= resourceCalls.size)
+            throw IllegalArgumentException("position is out of range of list")
+        resourceCalls.removeAt(position)
+    }
+
+    fun addResourceCall(position: Int, restCalls : RestResourceCalls) {
+        if(position > resourceCalls.size)
+            throw IllegalArgumentException("position is out of range of list")
+        resourceCalls.add(position, restCalls)
+    }
+
+    fun addResourceCall(restCalls : RestResourceCalls) {
+        resourceCalls.add(restCalls)
+    }
+
+    fun replaceResourceCall(position: Int, restCalls: RestResourceCalls){
+        if(position > resourceCalls.size)
+            throw IllegalArgumentException("position is out of range of list")
+        resourceCalls.set(position, restCalls)
+    }
+
+    fun swapResourceCall(position1: Int, position2: Int){
+        if(position1 > resourceCalls.size || position2 > resourceCalls.size)
+            throw IllegalArgumentException("position is out of range of list")
+        if(position1 == position2)
+            throw IllegalArgumentException("It is not necessary to swap two same position on the resource call list")
+        val first = resourceCalls[position1]
+        resourceCalls.set(position1, resourceCalls[position2])
+        resourceCalls.set(position2, first)
+    }
+
+
+    fun repairDBActions(sqlInsertBuilder: SqlInsertBuilder?){
+        val previousDbActions = mutableListOf<DbAction>()
+
+        getResourceCalls().forEach {
+            if (it.dbActions.isNotEmpty()){
```

Comment:
this could be a `filter` before the `forEach`, isn't it?

Reply:
fixed. thanks.